### PR TITLE
Implement HW4.

### PR DESCRIPTION
**Pull request by naztekinalp**

### CODE REVIEW

File: HW4/Bonsai.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package HW4;
+
+public class Bonsai extends HousePlant implements Trimmable {
+
+    String preferedPot = "shallow-small";
+    static int price = 31556;
```
Comment (drevalanya) [inline HW4/Bonsai.java]: unused fields
remove all unused code from the hw

File: HW4/Greenhouse.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
+package HW4;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Greenhouse {
```
Comment (drevalanya) [inline HW4/Greenhouse.java:7]: missed new line
do Ctrl+Shift+L in each class

File: out/production/NazTekinalp/HW3/Book.class
Comment (drevalanya) [inline out/production/NazTekinalp/HW3/Book.class:1]: Why it is here?

File: HW4/Main.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+        greenhouse.waterPlants();
+
+        greenhouse.changeTemperature(35);
+
+        greenhouse.removeAPlant(1);
+
+        greenhouse.retrievePlantsFromJP();
+
+
```
Comment (drevalanya) [inline HW4/Main.java]: Missed requirement from hw - add specific methods for Shrub...

File: HW4/Flowerable.java [hunk 4]
```diff
@@ -0,0 +1,5 @@
+package HW4;
+
+public interface Flowerable {
+    void flower();
```
Comment (drevalanya) [inline HW4/Flowerable.java]: unused method

File: HW4/FloweringPlant.java [hunk 5]
```diff
@@ -0,0 +1,13 @@
+package HW4;
+
+public abstract class FloweringPlant extends Plant implements Flowerable {
```
Comment (drevalanya) [inline HW4/FloweringPlant.java:3]: unused class. use it

File: HW4/Greenhouse.java (hunk 2 above)
Comment (drevalanya) [reply HW4/Greenhouse.java:7]: still no new line here

File: HW4/Greenhouse.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
+package HW4;
+
+import java.util.ArrayList;
+
+import java.util.List;
+
+public class Greenhouse {
+    int balance = 10000;
```
Comment (drevalanya) [inline HW4/Greenhouse.java]: unused field

File: out/production/NazTekinalp/.gitignore
Comment (drevalanya) [inline out/production/NazTekinalp/.gitignore:1]: why this everything still here?

**Review by drevalanya — COMMENTED** [review]
I mentioned not all places where unused code still available. Find them and use it

File: out/production/NazTekinalp/.gitignore
Comment (naztekinalp) [reply out/production/NazTekinalp/.gitignore:1]: out removed

File: HW4/Greenhouse.java (hunk 2 above)
Comment (naztekinalp) [reply HW4/Greenhouse.java:7]: updated

File: HW4/Greenhouse.java (hunk 6 above)
Comment (naztekinalp) [reply HW4/Greenhouse.java]: removed this field

File: HW4/FloweringPlant.java (hunk 5 above)
Comment (naztekinalp) [reply HW4/FloweringPlant.java:3]: used

**Review by drevalanya — APPROVED** [review]

**Pull request merged**